lib/Raven/SerializerInterface.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+
+namespace Raven;
+
+interface SerializerInterface
+{
+    /**
+     * Serialize an object (recursively) into something safe for data
+     * sanitization and encoding.
+     *
+     * @param mixed $value
+     * @param int   $max_depth
+     * @param int   $_depth
+     * @return string|bool|double|int|null|object|array
+     */
+    public function serialize($value, $max_depth = 3, $_depth = 0);
```

Comment:
Fortunately, as we're working on `2.x` branch we can break compatibility with older versions, so there shouldn't be any problem in doing it. Plus, I think it was a really big mistake to have all properties/methods public in the `1.x` version. We should expose only things that actually serves a purpose and I don't think that a public `serialize` method is so useful since the scope of this lbrary is not to be a serializer. So, here what I would do: make the `Serializer::serializeObject` protected (there is no reason to let it public), change the signature of the `Serializer::serialize` method according to the previously suggested definition (yes, breaking compatibility) and rework the code calling it to use the `$context` argument to pass the old arguments like `$max_depth` as key/values of the array. Plus, the `$depth` argument is something related to internal behaviour and should not be exposed anyway in a public method. One final though: should we provide a basic implementation of a serializer or should we just delegate it to a 3rd party library? I mean, there are a lot of cases to handle (circular references, how to cast objects/closures/different types) and there are already existing projects that do it well. Why should we reinvent the wheel?